bowtie/_report.py

```diff
@@ -327,17 +334,38 @@ def generate_badges(self, target_dir: Path, dialect: str):
                 - counts.skipped_tests
             )
             pct = (passed / total) * 100
-            impl_dir = target_dir / f"{lang}-{name}"
-            impl_dir.mkdir(parents=True, exist_ok=True)
             r, g, b = 100 - int(pct), int(pct), 0
-            badge = {
+            badge_per_draft = {
                 "schemaVersion": 1,
                 "label": label,
                 "message": "%d%% Passing" % int(pct),
                 "color": f"{r:02x}{g:02x}{b:02x}",
             }
-            badge_path = impl_dir / f"{label.replace(' ', '_')}.json"
-            badge_path.write_text(json.dumps(badge))
+            badge_supported_draft = {
```

Comment:
It would definitely be clearer, at least to me, if you grouped this by badge, not by functionality -- right now you do some calculation for a bunch of badges, then build a dict of bunch of badges, then write out a bunch of badges to files.

It's a lot clearer to me understanding I think to fully handle each badge one by one -- so calculate, build, write the first badge, then the second. And it may even then be clear there's some helper function that makes it even clearer. But I'm going to read this more carefully to remind myself what the existing badge means.

Reply:
The merge conflicts are because each time you're pushing, you're failing the linter (which is checking your code style), and so you're getting a commit that fixes that automatically.

There's a few ways to fix it, either look at what's being fixed and avoid doing that :) (in this case it's leaving whitespace on lines I think), or [install `pre-commit`](https://pre-commit.com/#install) which is what's being run, and then run that locally whenever you commit (it can do so automatically if you run `pre-commit install` inside the Bowtie repository, then it will run every time you commit).

Or alternatively, you can run `git pull` before you commit, and that will pull the commit that exists each time you're pushing.

Reply:
Okay, that helps a lot, I'll do that.